dts/arm/st/f0/stm32f091.dtsi

```diff
@@ -4,73 +4,60 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#include <st/f0/stm32f0.dtsi>
+#include <st/f0/stm32f071.dtsi>
 
 / {
 	soc {
-		clocks {
-			clk_hsi48: clk-hsi48 {
-				#clock-cells = <0>;
-				compatible = "fixed-clock";
-				clock-frequency = <DT_FREQ_M(48)>;
-				status = "disabled";
-			};
-		};
-
-		flash-controller@40022000 {
-			flash0: flash@8000000 {
-				erase-block-size = <2048>;
-			};
-		};
-
-		timers2: timers@40000000 {
-			compatible = "st,stm32-timers";
-			reg = <0x40000000 0x400>;
-			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00000001>;
-			resets = <&rctl STM32_RESET(APB1, 0U)>;
-			interrupts = <15 0>;
-			interrupt-names = "global";
-			st,prescaler = <0>;
+		/*
+		 * USARTs 3-8 share the same IRQ on stm32f091xx devices. This
+		 * configuration is not currently supported, so at most one of
+		 * these may be enabled at a time. Enabling more than one will
+		 * result in a build failure.
+		 */
+		usart5: serial@40005000 {
```

Comment:
This looks like an addition rather than a reorganization. A dedicated commit is required to ease the review.